File: 10.test_port/t.py
import socket

# 设置监听的端口
PORT = 12345

# 创建套接字并开始监听指定端口
server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
server.bind(('0.0.0.0', PORT))
server.listen(1)

print(f"Server started on port {PORT}")

# 接受连接
client, addr = server.accept()
print(f"Client connected from {addr}")

# 发送数据给客户端
client.send(b"Hello, this is the server!\n")

# 关闭连接
client.close()
server.close()

File: 10.test_port/t.lua

#lua t.lua

-- Lua script to create a simple server and test if the port is open

local socket = require("socket")

-- 设置监听的端口
local PORT = 12345

-- 创建服务器并开始监听指定端口
local server = socket.tcp()
server:setoption("reuseaddr", true)
server:bind("*", PORT)
server:listen()

print("Server started on port " .. PORT)

-- 等待连接
local client = server:accept()

if client then
    print("Client connected")
    client:send("Hello, this is the server!\n")
    client:close()
else
    print("No client connection")
end

-- 关闭服务器
server:close()
